### DISCUSSION TITLE
Distributed Tracing is not working with consul

### DISCUSSION THREAD

**Discussion by haider665** [apache/skywalking#12455] [Q&A]
I have two spring boot microservises **Service_A** and **Service_B** and used **Consul** as service discovery. There are two communication channel from **Service_A** to **Service_B**. One is **RabbitMQ** and another one is **http** call using **WebClient**. 

When, **Service_A** produce a message and **Service_B** consume that one, it shows in one trace. I checked the log and found the same trace id in both service.

But when I sent an **http** request from **Service_A** to **Service_B** using **Webclient** it should also show it in one trace but it shows two separate trace. I also checked the log of each service. It shows two different trace id.

Surprisingly, whey I use `webClientBuilder.baseUrl("http://localhost:port")` instead of `webClientBuilder.baseUrl("http://service-name")` it works perfectly. 

I have used these two gradle dependencies in my projects

```
    implementation 'org.apache.skywalking:apm-toolkit-trace:9.2.0'
    implementation 'org.apache.skywalking:apm-toolkit-logback-1.x:9.2.0'
```

Do I need to add any additional configuration in **webclient** while using **consul**?

**Comment by wu-sheng** [comment]
Trace continues or not, that depends on
- Whether new async machine is added in your codes or in the framework(webclient)
- Whether there is a gateway or something stops the header(key=sw8) to propagate.
